### DISCUSSION TITLE
FR: New Matrix "Mixed" view mode

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#15053] [Ideas]
The ability to set Matrix view modes in Craft 5 is an amazing AX enhancement. One way to take this one step further would be to add a **"mixed" view mode**. 

This would allow us to set the desired view mode _per entry type on a given matrix field_, with "card" and "inline block" modes available for each entry type. 

Let's say we have a matrix field with the following entry types: Text, Carousel, Team Members, Media Gallery, Two Column Section. 

- If we set the view mode to _inline_ we will potentialy have a very long editing form with a lot of "field noise" and too much information at the same time.
- If we set the view mode to _cards_, we will have a clean and compact form, but not immediate overview and editability of text content.
- With _mixed_ mode we could set only the Text and Two Column Section as inline blocks, and all others as cards. This would provide a way to have the entire text content easily glanceable and editable, while keeping the more complex parts tucked away in slideouts.

This is just an example but IMO a mixed view mode would bridge the gap nicely between traditional Matrix and CK Editor with embedded entries. Opinions more than welcome, as always. 🙂

**Comment by mmikkel** [comment]
I was just thinking about this today. It'd be amazing if there was a way to have some entry types render as inline-editable blocks, and other types as cards, all within the same field. 

UI wise, one way to implement something like this would be to make Matrix fields' entry type selector component into something more like a mini-FLD, with the ability to have "settings" per entry type (similar to how fields have certain field layout settings available in field layout designers), and with "View mode" being one of those settings. Something like that would also open up the possibility to add things like [entry type re-labelling](https://github.com/craftcms/cms/discussions/14779), conditional entry types, etc.

Potentially the Matrix field's "View Mode" setting would have to be renamed "Default View Mode" or something, but...

**Reply by thupsi** [reply]
That's exactly the way I would envision the UI. 👍

**Comment by shifuma** [comment]
I came here to request the same thing, and made a little mockup of how I saw it. Is this in line with what you were both thinking @mmikkel & @thupsi?

![view-mode](https://github.com/craftcms/cms/assets/86767543/d8f29f34-a1ef-403b-b98c-b4346a30c846)

![mixed-blocks](https://github.com/craftcms/cms/assets/86767543/610d1f28-b439-4e25-ad9b-939e45144aad)

**Reply by thupsi** [reply]
Pretty close, yes 👍

The only caveat is that this "override" should not be allowed when the matrix field view mode is set to "element index". So, maybe, it would be best to add a "mixed view" mode to explicitly allow overrides.

**Reply by thupsi** [reply]
Also, I think @mmikkel meant that the entry type slideout would be scoped to the specific matrix field and contain only override fields and conditionals. IMO that would cover more use cases.

**Reply by mmikkel** [reply]
Yeah, in my opinion the ideal thing would be a "Default View Mode" setting on the Matrix field, and then the ability to override that default per entry type, for that specific field.

Being able to set a view mode "globally" on the entry types themselves is an interesting idea, but in my opinion that would be much more limiting. In general, I think the "best" view mode for any particular entry type is pretty much always a contextual thing.

**Comment by brandonkelly** [comment]
I’ve considered something sortof similar to this: displaying all entries as cards initially, with “Edit” buttons that would swap them out with an inline-editable block.

**Reply by thupsi** [reply]
Hmm, what would be the advantage of this compared to opening in a slideout? Unless I'm misunderstanding.

**Reply by mmikkel** [reply]
IMO it's a really good idea. Compared to blocks, cards tend to offer a more compressed UI that can be quicker to scan (and reorder, disable, enable, etc). But on the flip side, inline-editable blocks feel more naturally "embedded" in the entry, and are typically the more intuitive interface when creating or editing - for example – "page builder"-like content and the like.

A view mode that rendered existing entries as cards, but let them be created and edited as inline blocks, would in certain use cases really feel like a best-of-both-worlds kind of thing.

That said – and regardless of a new view mode like this potentially being added – I still think the ability to use different view modes for different entry types within the same Matrix field is a really good idea, because not all content is the same, even if it's created and edited within the same field 🙂

**Reply by thupsi** [reply]
My worry is if it becomes the only option, or if it 'replaces' the FR discussed here. One would lose the ability to immediately find some text content that would be better served by a default inline view. On the other hand,  it's a bit like expanding and collapsing matrix blocks, but with a better preview when collapsed, which is a good thing.

**Reply by mmikkel** [reply]
Agreed – IMO this FR would absolutely still be relevant even if a new view mode like the one Brandon suggests is introduced. But I think a cards/blocks combo mode would address _some_ of what I feel this FR is about, and would possibly be a bit easier to implement – so potentially it'd be a good first step towards _ultimate view mode flexibility_? 🙂

**Reply by mmikkel** [reply]
Potentially, a cards/blocks combo view mode could also be a way to address https://github.com/craftcms/cms/discussions/14603

**Comment by ryanleichty** [comment]
In addition or along with this, it would be great if this could also be controlled by the user (or if there was a setting to enable control by users).

**Reply by humandevs** [reply]
From a Composable/Reusable standpoint, moving to Entries fields from Matrix blocks has allowed us to move to re-usable Entry blocks, instead of being stuck in a matrix block that can't be re-used outside that Matrix. 

However, the ease of inline editing of Matrix-in-Matrix blocks is superior to cards + flyout, so having the option to default specific entry type cards to inline editable (instead of flyout) would give the Composability/Reusability of re-usable Entries blocks, but the inline editing experience of matrix and matrix-in-matrix block editing.

**Comment by alexanderbuergin** [comment]
+1
